src/Catalyst.Core.Modules.Dfs/DfsService.cs

```diff
@@ -0,0 +1,529 @@
+                async () =>
+                {
+                    _stopTasks.Add(async () => await DhtService.StopAsync().ConfigureAwait(false));
+                    await DhtService.StartAsync().ConfigureAwait(false);
+                },
+                async () =>
+                {
+                    _stopTasks.Add(async () => await PingService.StopAsync().ConfigureAwait(false));
+                    await PingService.StartAsync().ConfigureAwait(false);
+                },
+                async () =>
+                {
+                    _stopTasks.Add(async () => await PubSubService.StopAsync().ConfigureAwait(false));
+                    await PubSubService.StartAsync().ConfigureAwait(false);
+                }
+            };
+
+            Log.Debug("waiting for services to start");
+            await Task.WhenAll(tasks.Select(t => t())).ConfigureAwait(false);
+
+            // Starting listening to the swarm.
+            var json = await ConfigApi.GetAsync("Addresses.Swarm").ConfigureAwait(false);
+            var numberListeners = 0;
+            foreach (string a in json)
+            {
+                try
+                {
+                    await SwarmService.StartListeningAsync(a).ConfigureAwait(false);
+                    ++numberListeners;
+                }
+                catch (Exception e)
+                {
+                    Log.Warning($"Listener failure for '{a}'", e);
+                }
+            }
+
+            if (numberListeners == 0)
+            {
+                Log.Error("No listeners were created.");
+            }
+
+            // Now that the listener addresses are established, the discovery 
+            // services can begin.
+            MulticastService multicast = null;
+            if (!Options.Discovery.DisableMdns)
+            {
+                multicast = new MulticastService();
+#pragma warning disable CS1998
+                _stopTasks.Add(async () => multicast.Dispose());
+#pragma warning restore CS1998
+            }
+
+            var autodialer = new AutoDialer(SwarmService)
+            {
+                MinConnections = Options.Swarm.MinConnections
+            };
+#pragma warning disable CS1998
+            _stopTasks.Add(async () => autodialer.Dispose());
+#pragma warning restore CS1998
+
+            tasks = new List<Func<Task>>
+            {
+                // Bootstrap discovery
+                async () =>
+                {
+                    var bootstrap = new Bootstrap
+                    {
+                        Addresses = await BootstrapApi.ListAsync()
+                    };
+                    bootstrap.PeerDiscovered += OnPeerDiscovered;
+                    _stopTasks.Add(async () => await bootstrap.StopAsync().ConfigureAwait(false));
+                    await bootstrap.StartAsync().ConfigureAwait(false);
+                },
+
+                async () =>
+                {
+                    if (Options.Discovery.DisableRandomWalk)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)